control-plane/msp-operator/src/main.rs

```diff
@@ -1060,13 +1060,18 @@ async fn main() -> anyhow::Result<()> {
 /// Normalize the disks if they have a schema, we dont want to change anything
 /// or do any error checking -- the loop will converge to the error state eventually
 fn normalize_disk(disk: &str) -> String {
-    Url::parse(disk).map_or(disk.to_string(), |u| {
+    let s = Url::parse(disk).map_or(disk.to_string(), |u| {
         u.to_file_path()
             .unwrap_or_else(|_| disk.into())
             .as_path()
             .display()
             .to_string()
-    })
+    });
+
+    match s.match_indices('?').next() {
```

Comment:
The URL crate provides a nice interface to the whole URL; perhaps it is better/easier to use those?